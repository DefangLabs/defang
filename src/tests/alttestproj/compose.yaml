name: tests
services:
  dfnx:
    restart: unless-stopped
    build:
      context: .
      dockerfile: Dockerfile
      target: testproj
      args:
        DNS: dfnx
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart
    x-defang-dns-role: arn:aws:iam::123456789012:role/ecs-service-role
    x-defang-static: ./static

secrets:
  dummy:
    external: true
    name: dummy

x-unsupported: unsupported
