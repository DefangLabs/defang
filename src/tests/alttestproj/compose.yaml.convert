[
  {
    "name": "dfnx",
    "platform": 2,
    "internal": true,
    "deploy": {
      "replicas": 1,
      "resources": {
        "reservations": {
          "memory": 256,
          "cpus": 0.25
        }
      }
    },
    "ports": [
      {
        "target": 80,
        "mode": 1
      },
      {
        "target": 1234,
        "mode": 1
      }
    ],
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "args": {
        "DNS": "dfnx"
      },
      "target": "alttestproj"
    },
    "healthcheck": {
      "test": [
        "CMD",
        "curl",
        "-f",
        "http://localhost/"
      ]
    },
    "networks": 1
  }
]