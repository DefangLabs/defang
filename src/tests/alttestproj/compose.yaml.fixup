[
  {
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile"
    },
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {
        "reservations": {
          "memory": "268435456"
        }
      },
      "placement": {}
    },
    "entrypoint": null,
    "networks": {
      "default": null
    },
    "restart": "unless-stopped"
  }
]