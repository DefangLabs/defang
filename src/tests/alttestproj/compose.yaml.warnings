 * Packaging the project files for dfnx at .