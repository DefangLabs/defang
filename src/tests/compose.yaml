name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
