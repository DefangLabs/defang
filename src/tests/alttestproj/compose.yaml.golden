name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: dfnx
      target: alttestproj
    deploy:
      resources:
        limits:
          cpus: 0.5
          memory: "536870912"
        reservations:
          cpus: 0.25
          memory: "268435456"
    healthcheck:
      test:
        - CMD
        - curl
        - -f
        - http://localhost/
    networks:
      default: null
    ports:
      - mode: ingress
        target: 80
      - target: 1234
    restart: unless-stopped
networks:
  default:
    name: tests_default
